api/routes/books.js

```diff
@@ -1,78 +1,153 @@
-    res.json({ status, response })
-  } catch (e) {
-    console.error(e)
-    const message = 'Failure to create. Please check request body and try again.'
-    const error = new Error(message)
-    error.status = 400
-    next(error)
-  }
-})
-
-// You should only be able to reserve a book if a user is logged in
-router.patch('/:id/reserve', async (req, res, next) => {
-  const { id } = req.params
-  try {
-    const book = await Book.findById(id)
-    if (!book) {
-      const error = new Error(`Invalid Book _id: ${id}`)
-      error.message = 404
-      return next(error)
-    }
-
-    book.reserved.status = true
-    // Set the reserved memberId to the current user
-    await book.save()
-    
-    const response = await Book.findById(book._id).select('-__v')
-    const status = 200
-    res.json({ status, response })
-  } catch (e) {
-    console.error(e)
-  }
-})
-
-// You should only be able to return a book if the user is logged in
-// and that user is the one who reserved the book
-router.patch('/:id/return', async (req, res, next) => {
-  const status = 200
-  const message = 'You must implement this route!'
-  
-  console.log(message)
-  res.status(status).json({ status, message })
-})
-
+const router = require('express').Router()
+const Book = require('../models/book')
+const User = require('../models/user');
+const jsonwebtoken = require('jsonwebtoken');
+
+const { SECRET_KEY } = process.env;
+
+router.get('/', async (req, res, next) => {
+  const status = 200;
+  const response = await Book.find().select('-__v');
+  res.json({ status, response });
+})
+
+router.get('/:id', async (req, res, next) => {
+  const { id } = req.params;
+  const status = 200
+  try {
+    const response = await Book.findById(id).select('-__v');
+    if (!response) throw new Error(`Invalid Book _id: ${id}`);
+    res.json({ status, response });
+  } catch (e) {
+    console.error(e);
+    const error = new Error(`Cannot find book with id ${id}.`);
+    error.status = 404;
+    next(error);
+  }
+})
+
+/***
+ * - [ ] **Update the `POST /api/books` route:** This route should only be available to users who are admins. 
+ * If the user is an admin, proceed as normal. If they are not an admin, return an error message with a status code of 401.
+ */
+
+router.post('/', async (req, res, next) => {
+  const status = 200;
+  try {
+    const token = req.headers.authorization.split(`Bearer `)[1];
+    const isValid = jsonwebtoken.verify(token, SECRET_KEY);
+    if (!isValid) {
+        throw new Error(`Token is not valid. Please check your request and try again.`);
+        //error.message = 401;   I continously ran into failed requests until i commented this out and changed the above from `const error = new Error...` to the current statement. Any suggestion as to why this is?
+        //return next(error);
+    }
+    const isAdmin = await User.findOne({_id: isValid.id});
+    if (isAdmin.admin = false) {

```

Comment:
This one is actually a different issue. Here, you're only using one `=` which means you're _assigning_ the value of `false` to `isAdmin.admin`.